src/spdx_matcher/__init__.py

```diff
@@ -206,142 +214,205 @@ def normalize(license_text, remove_sections=REMOVE_FINGERPRINT):
+                    json_detail_data["licenseExceptionText"],
+                    fast_exit=False,
                 )
-            if "licenseExceptionText" in json_detail_data:
-                if "regexpForMatch" in json_detail_data:
-                    start_time = time.time()
-                    match, license_data, full_match = _license_regexps_match(
-                        json_detail_data["regexpForMatch"],
-                        json_detail_data["licenseExceptionText"],
-                        fast_exit=False,
+                execution_time = time.time() - start_time
+                json_detail_data["matchCost"] = execution_time
+                json_detail_data["matchConfidence"] = match
+                if match < 0.99:
+                    print(
+                        f"Unable to match {match} full_match:{full_match}  exmplar text "
+                        f"to regexp for exception "
+                        f"{json_detail_data['licenseExceptionId']}  "
+                        f"license_data {json.dumps(license_data)} time {execution_time}"
                     )
-                    execution_time = time.time() - start_time
-                    json_detail_data["matchCost"] = execution_time
-                    json_detail_data["matchConfidence"] = match
-                    if match < 0.99:
-                        print(
-                            f"Unable to match {match} full_match:{full_match}  exmplar text "
-                            f"to regexp for exception "
-                            f"{json_detail_data['licenseExceptionId']}  "
-                            f"license_data {json.dumps(license_data)} time {execution_time}"
-                        )
-                    else:
-                        print(
-                            f"Success to match and full_match:{full_match} exmplar text to "
-                            f"regexp for exception {json_detail_data['licenseExceptionId']} "
-                            f"license_data {json.dumps(license_data)} time {execution_time}"
-                        )
-            for k in json_detail_data:
-                if k not in exception:
-                    exception[k] = json_detail_data[k]
-            match_cache["exceptions"][exception["licenseExceptionId"]] = {
+                else:
+                    print(
+                        f"Success to match and full_match:{full_match} exmplar text to "
+                        f"regexp for exception {json_detail_data['licenseExceptionId']} "
+                        f"license_data {json.dumps(license_data)} time {execution_time}"
+                    )
+        for k in json_detail_data:
+            if k not in exception:
+                exception[k] = json_detail_data[k]
+        return {
+            exception["licenseExceptionId"]: {
                 "name": exception["name"],
+                "licenseText": exception["licenseExceptionText"],
```

Comment:
Add the source text for jaro similarity, I try to connect the finger pints into one single string and calculate the score of similarity, but the result is not good, so I add the source license text from spdx license data to make it more correct

Reply:
That makes sense